pkg/apis/core/v1beta1/helper/helpers_test.go

```diff
@@ -595,365 +663,427 @@ var _ = Describe("helper", func() {
+				return
 			}
-
-			_, _, err := GetShootMachineImageFromLatestMachineImageVersion(inputImage)
-			Expect(err).To(HaveOccurred())
-		})
-	})
-
-	var (
-		kubernetesSettings = gardencorev1beta1.KubernetesSettings{
-			Versions: []gardencorev1beta1.ExpirableVersion{
-				{Version: "1.15.1"},
-				{Version: "1.14.4"},
-				{Version: "1.12.9"},
-			},
-		}
-	)
-
-	DescribeTable("#DetermineLatestKubernetesPatchVersion",
-		func(cloudProfile gardencorev1beta1.CloudProfile, currentVersion, expectedVersion string, expectVersion bool) {
-			ok, newVersion, err := DetermineLatestKubernetesPatchVersion(&cloudProfile, currentVersion)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(ok).To(Equal(expectVersion))
-			Expect(newVersion).To(Equal(expectedVersion))
+			Expect(qualifyingVersionFound).To(Equal(expectVersionToBeFound))
+			Expect(latestVersion).To(Equal(expected))
 		},
-
-		Entry("version = 1.15.1",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
-				},
-			},
-			"1.15.0",
-			"1.15.1",
-			true,
-		),
-
-		Entry("version = 1.12.9",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
+		Entry("Get latest version",
+			gardencorev1beta1.MachineImage{
+				Name: "gardenlinux",
+				Versions: []gardencorev1beta1.ExpirableVersion{
+					{
+						Version: "1.17.1",
+					},
+					{
+						Version: "1.15.0",
+					},
+					{
+						Version: "1.14.3",
+					},
+					{
+						Version: "1.13.1",
+					},
 				},
 			},
-			"1.12.4",
-			"1.12.9",
 			true,
-		),
-
-		Entry("no new version",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
-				},
+			gardencorev1beta1.ShootMachineImage{
+				Name:    "gardenlinux",
+				Version: pointer.StringPtr("1.17.1"),
 			},
-			"1.15.1",
-			"",
 			false,
 		),
-	)
-
-	DescribeTable("#DetermineNextKubernetesMinorVersion",
-		func(cloudProfile gardencorev1beta1.CloudProfile, currentVersion, expectedVersion string, expectVersion bool) {
-			ok, newVersion, err := DetermineNextKubernetesMinorVersion(&cloudProfile, currentVersion)
-
-			Expect(err).ToNot(HaveOccurred())
-			Expect(ok).To(Equal(expectVersion))
-			Expect(newVersion).To(Equal(expectedVersion))
-		},
-
-		Entry("version = 1.15.1",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
-				},
-			},
-			"1.14.4",
-			"1.15.1",
-			true,
-		),
-
-		Entry("version = 1.12.9",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
-				},
-			},
-			"1.11.0",
-			"1.12.9",
-			true,
-		),
-
-		Entry("no new version",
-			gardencorev1beta1.CloudProfile{
-				Spec: gardencorev1beta1.CloudProfileSpec{
-					Kubernetes: kubernetesSettings,
+		FEntry("Expect error, because no latest version could be found",
```

Comment:
`cd pkg && ginkgo unfocus`.
```suggestion
		Entry("Expect error, because no latest version could be found",
```

Reply:
thanks